### PR TITLE
docs/library/uos.rst Clarify AbstractBlockDev section

### PR DESCRIPTION
**Pull request by peterhinch**
Implements the change requested in #5478 

The statement in the last para on `ioctl` op 6 is based on experiment and existing code samples. I'd like to clarify the meaning of the return value if someone can provide guidance.

### CODE REVIEW

File: docs/library/uos.rst [hunk 1]
```diff
@@ -218,11 +218,19 @@ represented by VFS classes.
...
-functionality a piece of hardware (like flash memory).  A block device can be
-used by a particular filesystem driver to store the data for its filesystem.
+A block device is an object which implements the block protocol. This enables a
+device to support MicroPython filesystems. The physical hardware is represented
+by a user defined class. The :class:`AbstractBlockDev` class is a template for
+the design of such a class: MicroPython does not actually provide that class,
+but an actual block device class must implement the methods described below.
+
+A concrete implementation of this class will usually allow access to the
+memory-like functionality a piece of hardware (like flash memory). A block
```
Comment (tve) [inline docs/library/uos.rst]: "a piece of hardware" -> "of a piece of hardware"

File: docs/library/uos.rst [hunk 2]
```diff
@@ -294,5 +302,7 @@ that the block device supports the extended interface.
             (*arg* is unused)
           - 6 -- erase a block, *arg* is the block number to erase
 
-See :ref:`filesystem` for example implementations of block devices using both
-protocols.
+Unless otherwise stated ``ioctl(op, arg)`` can return None. Consequently an
+implementation can ignore unused values of ``op``. As a minimum value 4 must be
```
Comment (tve) [inline docs/library/uos.rst]: "As a minimum value 4" -> "As a minimum, value 4" (or As a minimum `ioctl(4,...)`", I was reading "minimum value" and that didn't make sense, took me while to understand that you mean "minimum implementation".

**Review by tve — COMMENTED** [review]
Small nitpicks.

**Comment by peterhinch** [comment]
Thank you, will do.

File: docs/library/uos.rst [hunk 3]
```diff
@@ -294,5 +302,7 @@ that the block device supports the extended interface.
             (*arg* is unused)
           - 6 -- erase a block, *arg* is the block number to erase
 
-See :ref:`filesystem` for example implementations of block devices using both
-protocols.
+Unless otherwise stated ``ioctl(op, arg)`` can return None. Consequently an
+implementation can ignore unused values of ``op``. As a minimum
+``ioctl(4, ...)`` must be intercepted; for littlefs ``ioctl(6, ...)`` must be
+intercepted and return 0. The need for others is hardware dependent.
```
Comment (dpgeorge) [inline docs/library/uos.rst]: Perhaps this para would be better indented so it's part of the ioctl method description?

So the intended semantics for the return value is:
- if it's not implemented then return `None` (so unused `op`'s can be safely ignored, as you point out)
- 4 and 5 should return ints (as already documented)
- others, if implemented, should return 0 for success and non-zero for failure, with the value returned being an OSError errno code

Comment (peterhinch) [reply docs/library/uos.rst]: Thank you for that. I've now updated the doc.

**Comment by dpgeorge** [comment]
Thank you!  Squashed and merged in 59746ac14a7e639b8abd50fe626549e115c19c68

**Pull request closed without merging**